client-js/app/elements/labrad-grapher.ts

```diff
@@ -35,30 +52,46 @@ export class LabradGrapher extends polymer.Base {
     this.showStars = false;
     this.showTrash = false;
     this.updateButtons();
+
+    this.initializeListItems_(
+      this.path,
+      this.dirs,
+      this.datasets);
+    this.applyListFilter_();
   }
 
   starClicked() {
     this.showStars = !this.showStars;
     if (this.showStars) this.showTrash = false;
     this.updateButtons();
-    this.kick++;
+    this.applyListFilter_();
   }
 
   trashClicked() {
```

Comment:
There is quite a bit more in this file I wanted to rename/refactor, but I was trying to keep this as small and single-function of a PR as possible.